src/sinks/util/http.rs

```diff
@@ -190,7 +192,7 @@ where
 
         request
             .headers_mut()
-            .insert("User-Agent", self.version.clone());
+            .insert("User-Agent", self.user_agent.clone());
```

Comment:
Is Vector's user agent inserted before or after user-defined headers from [`headers`](https://vector.dev/docs/reference/sinks/http/#headers) configuration option?

If it is inserted after them, I think it might make sense to check whether `User-Agent` header is already present in headers and insert the default one only if it is already not present.

The reason is that I think it should be possible for allow users to override `User-Agent` in `http` sink if they really want to.

Reply:
Totally valid use case! I'll make that adjustment.